traitsui/tests/editors/test_check_list_editor.py

```diff
@@ -315,92 +315,78 @@ def test_custom_editor_mapping_name_tuple(self):
             self.check_checklist_mappings_tuple_name_change("custom")
 
     def test_simple_editor_checklist_values_change_dispose(self):
-        if is_current_backend_wx():
-            # Missing cleanup in wx (enthought/traitsui#752)
-            with self.assertRaises(AttributeError):
-                self.check_checklist_values_change_after_ui_dispose("simple")
-        else:
-            self.check_checklist_values_change_after_ui_dispose("simple")
+        self.check_checklist_values_change_after_ui_dispose("simple")
 
     def test_custom_editor_checklist_values_change_dispose(self):
-        if is_current_backend_wx():
-            # Missing cleanup in wx (enthought/traitsui#752)
-            with self.assertRaises(AttributeError):
-                self.check_checklist_values_change_after_ui_dispose("custom")
-        else:
-            self.check_checklist_values_change_after_ui_dispose("custom")
+        self.check_checklist_values_change_after_ui_dispose("custom")
 
 
 @skip_if_null
 class TestSimpleCheckListEditor(unittest.TestCase):
 
+    @contextlib.contextmanager
     def setup_gui(self, model, view):
-        gui = GUI()
-        ui = model.edit_traits(view=view)
-        self.addCleanup(ui.dispose)
-
-        gui.process_events()
-        editor = ui.get_editors("value")[0]
-        combobox = editor.control
-
-        return gui, editor, combobox
+        with create_ui(model, dict(view=view)) as ui:
+            process_cascade_events()
```

Comment:
Maybe. I am not sure these instances of `process_cascade_events` right after the ui is set up is really needed: Some of them could have been motivated by test interactions that are now resolved by `create_ui` flushing the event queue after dispose.  I would defer that to a separate PR because it requires a bit more understanding of the original tests.

Reply:
> Would it make sense to add this to create_ui instead of individual tests?

Just to clarify, these `process_cascade_events` are replacing the existing `GUI.process_events` as an attempt to keep existing code logic. In other words, they are not new.